arkouda/client.py

```diff
@@ -132,7 +127,373 @@ def set_defaults() -> None:
     maxTransferBytes = maxTransferBytesDefVal
 
 
-# create context, request end of socket, and connect to it
+class ChannelType(Enum):
+    """
+    The ChannelType Enum specifies which Channel implementation is
+    to be used for an Arkouda client deployment.
+    """
+    ZMQ = 'ZMQ'
+
+    def __str__(self) -> str:
+        """
+        Overridden method returns value.
+        """
+        return self.value
+
+    def __repr__(self) -> str:
+        """
+        Overridden method returns value.
+        """
+        return self.value
+
+
+class Channel():
+    """
+    The Channel class defines methods for connecting to and communicating with
+    the Arkouda server.
+    """
+    __slots__ = ('url', 'user', 'token', 'logger')
+
+    def __init__(self, user: str, server: str, port: int, token: str = None,
+                 connect_url: str = None) -> None:
+        self._set_url(server, port, connect_url)
+        self.user = user
+        self.token = self._get_access_token(server, port, token)
+        self.logger = getArkoudaLogger(name="Arkouda Client")
+
+    def _set_url(self, server: str, port: int, connect_url: str = None) -> None:
+        """
+        Generates a url per the Channel protocol as well as host and port and sets the
+        url to the generated value if connect_url is None. Otherwise, sets the url to
+        the connect_url value.
+
+        Parameters
+        ----------
+        server : str
+            Arkouda server hostname, ip address, or service name
+        port : int
+            Arkouda server host port
+        connect_url : str, o
+            The complete url in the format of tcp://server:port?token=<token_value>
+            where the token is optional
+
+        Returns
+        -------
+        None
+        """
+        self.url = connect_url if connect_url else f'tcp://{server}:{port}'
+
+    def _get_access_token(self, server: str, port: int, token: Optional[str]) -> Optional[str]:
+        """
+        Sets the access_token for the connect request by doing the following:
```

Comment:
Argh, will fix, thanks for catching that!

Reply:
@pierce314159 Fixed! Thanks for catching this. I set the method name to _set_access_token and refactored so that it sets the token instance attribute as opposed to returning the token string.